airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/auth/AuthButton.test.tsx

```diff
@@ -0,0 +1,148 @@
+import { screen, render } from "@testing-library/react";
+
+import { TestWrapper } from "utils/testutils";
+import { useServiceForm } from "views/Connector/ServiceForm/serviceFormContext";
+
+import { AuthButton } from "./AuthButton";
+import { useFormikOauthAdapter } from "./useOauthFlowAdapter";
+jest.setTimeout(10000);
+
+jest.mock("formik", () => {
+  const useFormikContext = () => {
+    const values = {};
+    const submitCount = 0;
+    return { values, submitCount };
+  };
+
+  return { useFormikContext };
+});
+
+/**
+ * Mock services to reuse multiple times with different values:
+ * 1. mock the service
+ * 2. cast the type as shown below.  `as unknown` is required to get the typing to play nice as there isn't crossover
+ *    Partial is optional, but required if you do not want to mock the entire object
+ *
+ * Then, can implement in tests using useWhateverServiceYouMocked.mockImplementationOnce or useWhateverServiceYouMocked.mockImplementation
+ */
+
+jest.mock("views/Connector/ServiceForm/components/Sections/auth/useOAuthFlowAdapter");
+const mockUseFormikOauthAdapter = useFormikOauthAdapter as unknown as jest.Mock<Partial<typeof useFormikOauthAdapter>>;
+const baseUseFormikOauthAdapterValues = {
+  run: jest.fn(),
+  loading: false,
+};
+
+jest.mock("views/Connector/ServiceForm/serviceFormContext");
+const mockUseServiceForm = useServiceForm as unknown as jest.Mock<Partial<typeof useServiceForm>>;
+const baseUseServiceFormValues = {
+  selectedConnector: "abcde",
+  allowOAuthConnector: true,
+  selectedService: undefined,
+};
+
+describe("auth button", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("it initially renders with correct message and no status message", () => {
+    // no auth errors
+    mockUseServiceForm.mockImplementationOnce(() => {
+      const authErrors = {};
+      const { selectedConnector, allowOAuthConnector, selectedService } = baseUseServiceFormValues;
```

Comment:
Added the `base` objects to keep repeated code out of these mock implementations so it was easier to tell which property _was_ changing with each implementation.